src/mscorlib/Common/System/SR.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Resources;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class SR
+    {
+        private static ResourceManager s_resourceManager;
+
+        private static ResourceManager ResourceManager
+        {
+            get
+            {
+                if (SR.s_resourceManager == null)
```

Comment:
There are a few places in the code where I couldn't replace a call to Environment.GetResourceString with the equivalent SR call because they use either computed or passed in resource names. Should I change those to use SR.GetResourceString?